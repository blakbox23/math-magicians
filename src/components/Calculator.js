import React, { Component } from 'react';
import '../App.css';

class Calculator extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div className="calc">
        <div className="screen">
          0
        </div>
        <div id="row1">
          <div id="c1r1">AC</div>
          <div id="c2r1">+/-</div>
          <div id="c3r1">%</div>
          <div id="c4r1" className="operators">/</div>
        </div>
        <div id="row2">
          <div id="c1r2">7</div>
          <div id="c2r2">8</div>
          <div id="c3r2">9</div>
          <div id="c4r2" className="operators">*</div>
        </div>
        <div id="row3">
          <div id="c1r3">4</div>
          <div id="c2r3">5</div>
          <div id="c3r3">6</div>
          <div id="c4r3" className="operators">-</div>
        </div>
        <div id="row4">
          <div id="c1r4">1</div>
          <div id="c2r4">2</div>
          <div id="c3r4">3</div>
          <div id="c4r4" className="operators">+</div>
        </div>
        <div id="row5">
          <div id="c1r5">0</div>
          <div id="c2r5">.</div>
          <div id="c3r5" className="operators">=</div>
        </div>
      </div>
    );
  }
}

export default Calculator;
