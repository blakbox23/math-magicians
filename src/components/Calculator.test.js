/* eslint-disable no-unused-expressions */
import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import Calculator from './Calculator';

describe('It should get calculator snapshot', () => {
  test('renders snapshot', () => {
    const component = renderer.create(<Calculator />).toJSON();
    expect(component).toMatchSnapshot('Calculator');
  });
});

describe('Check the click event', () => {
  test('click event', () => {
    render(<Calculator />);
    expect(userEvent.click(screen.getByTestId('ac'))).toHaveBeenCalled;
  });
});

