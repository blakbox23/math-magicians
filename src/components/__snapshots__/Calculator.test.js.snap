// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should get calculator snapshot renders snapshot: Calculator 1`] = `
<div
  className="calcPg"
>
  <h2>
    Lets do some Math!
  </h2>
  <div
    className="calc"
  >
    <div
      className="screen"
    >
      0
    </div>
    <div
      id="row1"
    >
      <div
        data-testid="ac"
        id="c1r1"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        AC
      </div>
      <div
        id="c2r1"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        +/-
      </div>
      <div
        id="c3r1"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        %
      </div>
      <div
        className="operators"
        id="c4r1"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        รท
      </div>
    </div>
    <div
      id="row2"
    >
      <div
        id="c1r2"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        7
      </div>
      <div
        id="c2r2"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        8
      </div>
      <div
        id="c3r2"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        9
      </div>
      <div
        className="operators"
        id="c4r2"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        x
      </div>
    </div>
    <div
      id="row3"
    >
      <div
        id="c1r3"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        4
      </div>
      <div
        id="c2r3"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        5
      </div>
      <div
        id="c3r3"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        6
      </div>
      <div
        className="operators"
        id="c4r3"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        -
      </div>
    </div>
    <div
      id="row4"
    >
      <div
        id="c1r4"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        1
      </div>
      <div
        id="c2r4"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        2
      </div>
      <div
        id="c3r4"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        3
      </div>
      <div
        className="operators"
        id="c4r4"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        +
      </div>
    </div>
    <div
      id="row5"
    >
      <div
        id="c1r5"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        0
      </div>
      <div
        id="c2r5"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        .
      </div>
      <div
        className="operators"
        id="c3r5"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        =
      </div>
    </div>
  </div>
</div>
`;
